hedged.go

```diff
@@ -53,40 +53,49 @@ func (ht *hedgedTransport) RoundTrip(req *http.Request) (*http.Response, error)
 	defer cancel()
 	req = req.WithContext(ctx)
 
-	var res interface{}
-	resultCh := make(chan interface{}, ht.upto)
+	var res *http.Response
+	var err error
+	resultCh := make(chan *http.Response, ht.upto)
```

Comment:
Looks like we're waiting for only 1 results, so the size of then channel can be 1, wdyt?

Reply:
Making resultCh with buffer size 1 can lead to leak of goroutines if more than one groutine got successful result.
We can still do it but wait for channel Done() as well.

Reply:
As I thought, so leaving as it's.